### PR TITLE
ManagedCluster and Klusterlet

### PR DESCRIPTION
**Pull request by IlonaShishov**
Signed-off-by: Ilona Shishov <Ilona.Shishov@gmail.com>

### CODE REVIEW

File: src/view/ClusterDetailsPanel.ts [hunk 1]
```diff
@@ -176,10 +178,16 @@ export class ClusterDetailsPanel {
 						// get managed cluster addons
 						managedClusterAddons = await this._kubeDataLoader.getResources(selectedCluster, "ManagedClusterAddOn");
 						this._panel.webview.postMessage({"managedClusterAddons":JSON.stringify(managedClusterAddons)});
+						// get managed cluster
+						clusterManager = await this._kubeDataLoader.getResources(selectedCluster, "ClusterManager");
+						this._panel.webview.postMessage({"clusterManager":JSON.stringify(clusterManager)});
 					} else {
```
Comment (ilan-pinto) [inline src/view/ClusterDetailsPanel.ts:184]: theoretically,  the same cluster can be a hub cluster and also managed cluster (by another hub) 
we should check out if klusterlet exists. 
at line 162.
then replace else with
`if(klusterlet.length > 0) { 
   show managed cluster resources  }
`

File: webview-ui/src/comp/Klusterlet.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+         } );          
+    },[])
+
+    return (  
+        <section className="component-row"> 
+            { klusterlet.length >0 &&  
+                <>
+                    <h2 style={{ marginTop: '40px' }}>Klusterlet</h2>        
+                    <VSCodeDataGrid gridTemplateColumns="1fr 1fr" aria-label='Klusterlet' >
+                        <VSCodeDataGridRow rowType="sticky-header">
```
Comment (ilan-pinto) [inline webview-ui/src/comp/Klusterlet.tsx:20]: klusterlet also holds the hub cluster name. Need to expose it.

**Review by ilan-pinto — CHANGES_REQUESTED** [review]

File: src/view/ClusterDetailsPanel.ts [hunk 3]
```diff
@@ -181,17 +181,27 @@ export class ClusterDetailsPanel {
 						// get managed cluster
 						clusterManager = await this._kubeDataLoader.getResources(selectedCluster, "ClusterManager");
 						this._panel.webview.postMessage({"clusterManager":JSON.stringify(clusterManager)});
-					} else {
+					}
+					try {
+						klusterlet = await this._kubeDataLoader.getResources(selectedCluster, "Klusterlet");
+
+					} catch (exceptionVar) {
+						klusterlet = [];
```
Comment (ilan-pinto) [inline src/view/ClusterDetailsPanel.ts:189]: for this version write exception to log

File: src/view/ClusterDetailsPanel.ts [hunk 4]
```diff
@@ -160,7 +160,7 @@ export class ClusterDetailsPanel {
 
 				if (selectedCluster.length > 0) {
 					managedClusters = await this._kubeDataLoader.loadManagedCluster(selectedCluster);
```
Comment (ilan-pinto) [inline src/view/ClusterDetailsPanel.ts:162]: Add try catch as done for klusterlet in line 185

File: src/view/ClusterDetailsPanel.ts [hunk 5]
```diff
@@ -181,17 +181,27 @@ export class ClusterDetailsPanel {
...
 						appliedManifestWork = await this._kubeDataLoader.getResources(selectedCluster, "AppliedManifestWork");
 						this._panel.webview.postMessage({"appliedManifestWork":JSON.stringify(appliedManifestWork)});
-						// get klusterlet
-						klusterlet = await this._kubeDataLoader.getResources(selectedCluster, "Klusterlet");
-						this._panel.webview.postMessage({"klusterlet":JSON.stringify(klusterlet)});
 					}
-					// get subscriptions
-					subscriptions = await this._kubeDataLoader.getResources(selectedCluster, "Subscription");
-					this._panel.webview.postMessage({"subscriptions":JSON.stringify(subscriptions)});
+					if (klusterlet.length !== 0 || managedClusters.length !== 0 ){
```
Comment (ilan-pinto) [inline src/view/ClusterDetailsPanel.ts:200]: Subscription is a hub or managed cluster resources?
I don't think it resides on both

**Review by ilan-pinto — CHANGES_REQUESTED** [review]

File: src/view/ClusterDetailsPanel.ts (hunk 1 above)
Comment (IlonaShishov) [reply src/view/ClusterDetailsPanel.ts]: fixed, Thank you

File: src/view/ClusterDetailsPanel.ts (hunk 3 above)
Comment (IlonaShishov) [reply src/view/ClusterDetailsPanel.ts]: try/catch removed and replaced by suitable return statement in kube.ts

File: src/view/ClusterDetailsPanel.ts (hunk 4 above)
Comment (IlonaShishov) [reply src/view/ClusterDetailsPanel.ts:162]: try/catch removed and replaced by suitable return statement in kube.ts

File: src/view/ClusterDetailsPanel.ts (hunk 5 above)
Comment (IlonaShishov) [reply src/view/ClusterDetailsPanel.ts:195]: Subscription resides in both Hub Cluster and Managed Clusters
in Hub Cluster - subscription status: Propagated
in Managed Cluster - subscription status: Subscribed

**Review by ilan-pinto — APPROVED** [review]

**Pull request merged**